### PR TITLE
Implement predicate

### PR DESCRIPTION
**Pull request by fanzhangio**
https://github.com/kubernetes-sigs/kubebuilder/issues/240

### CODE REVIEW

**Comment by pwittrock** [comment]
Why not copy the kubebuilder Predicates over:

https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/controller/predicates/predicates.go

The main ones we need are the ResourceVersionChanged, and the ability to compose predicates with And/Or

File: pkg/predicate/predicate.go [hunk 1]
```diff
@@ -16,7 +16,12 @@ limitations under the License.
 
 package predicate
 
-import "sigs.k8s.io/controller-runtime/pkg/event"
+import (
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var predicateLog = logf.KBLog.WithName("predicate").WithName("eventFilters")
```
Comment (droot) [inline pkg/predicate/predicate.go]: predicateLog --> log ?

Comment (fanzhangio) [reply pkg/predicate/predicate.go]: Don't we use unified `runtime log` ?

Comment (droot) [reply pkg/predicate/predicate.go]: I was suggesting to rename `predicateLog` --> `log`

**Pull request merged**